### ISSUE TITLE
Adapt Platform FE to upcoming Genetics Portal data

### ISSUE THREAD

**Issue by d0choa** [opentargets/platform#816]
Front-end changes necessary to allocate the Genetics Portal data in the Platform
Details TBC.

**Comment by andrewhercules** [comment]
I have created an iInitial mockup of the [Genetic Associations (Common Diseases) data table](https://docs.google.com/presentation/d/1qzbY5xCn6ut_XO-OFE0o58xZPurdqMeHCtf7LtRP8QI) and it is available on the team Google Drive. 

Based on a preliminary review by @DSuveges, @d0choa, and @edm1, a few questions remain open for discussion:

1. Can we obtain the reported trait information for GWAS Catalog and PheWAS Catalog evidence string data?
2. Can we obtain the author and year of the publication for GWAS Catalog and PheWAS Catalog evidence string data?  
3. For rows with L2G data, we can link the variant ID to the variant page on the Genetics Portal. However, for rows without L2G data, we could continue to link to Ensembl or can we determine if the RSID entry exists in the Genetics Portal?

In order to deploy the mockup into production, it will require:

- [ ] Disease EFO label
- [ ] Disease EFO ID
- [ ] Reported trait for L2G data (and possibly for GWAS Catalog and PheWAS Catalog data if available)
- [ ] Study ID (if available)
- [ ] Publication ID
- [ ] Publication author and year (low priority - a publication link is fine for an MVP)
- [ ] RSID
- [ ] Chromosome position annotation for variant
- [ ] Variant type
- [ ] p-value
- [ ] Odds ratio (confidence)
- [ ] Gene prioritisation score (either L2G score or association score for given evidence string)
- [ ] GWAS Catalog study URL
- [ ] PheWAS Catalog study URL

**Comment by DSuveges** [comment]
Some of these values are not yet exported by the GWAS team, I send an email to them so they can fix their pipeline. The newly added values are: 

* For the publication: year, first author
* For the disease: Reported trait
* For variant: Chromosome position annotation for variant

**Comment by andrewhercules** [comment]
Based on feedback from the team and a review of the data from the API ([example](https://platform-api-qc.opentargets.io/v3/platform/public/evidence/filter?size=1000&datasource=ot_genetics_portal&datasource=gwas_catalog&datasource=phewas_catalog&fields=unique_association_fields&fields=disease&fields=evidence&fields=variant&fields=target&fields=sourceID&fields=access_level&target=ENSG00000064687&disease=EFO_0000249&expandefo=true)), I have updated the data table design for the new L2G data - see below:

![Screenshot 2020-02-19 at 15 54 15](https://user-images.githubusercontent.com/7490258/74848969-1ab7a500-5330-11ea-9236-3b0e5cf82998.png)

@LucaFumis, here are the design notes for each column in the data table:

**Disease**
For both PheWAS data and the Open Targets Genetics Portal L2G data, please display `disease.efo_info.label` and hyperlink to the Platform disease profile page using the `disease.id` value.

**Reported trait name**
For PheWAS Catalog data: please display N/A.
For Open Targets Genetics Portal L2G data: please display `disease.reported_trait`

**Publications**
For both PheWAS data and the Open Targets Genetics Portal L2G data, please implement the existing functionality - number of publications badge and a link to EuropePMC

**Variant**
For PheWAS Catalog data: please implement the existing functionality of displaying the RSID and hyperlink to Ensembl.
For Open Targets Genetics Portal L2G data: please display `variant.rs_id` value and hyperlink to the Genetics Portal using the `variant.source_link` value.

**Variant type**
For both PheWAS data and the Open Targets Genetics Portal L2G data, please implement the existing functionality (note: please ignore the SO_0001060 issue which will be resolved in the rewrite)

**Association p-value**
For PheWAS Catalog data: please implement the existing functionality of displaying the p-value using the `evidence.variant2disease.resource_score.value` value.
For Open Targets Genetics Portal L2G data: please display the p-value as a string using the `evidence.variant2disease.resource_score.mantissa` + e + `evidence.variant2disease.resource_score.exponent`.

**Odds ratio (confidence)**
Please drop this column.

**Gene prioritisation score**
For PheWAS Catalog data: please display N/A.
For Open Targets Genetics Portal L2G data: please display the L2G pipeline score in `evidence.gene2variant.resource_score.value`.

**Evidence source**
For PheWAS Catalog data: please implement the existing functionality of displaying PheWAS Catalog and linking to their website.
For Open Targets Genetics Portal L2G data: please display `Open Targets Genetics Portal` and hyperlink to the study-locus page in the Genetics Portal - for example https://genetics.opentargets.org/study-locus/GCST004610/19_1026478_G_T. To construct this link, please extract the study ID from `evidence.variant2disease.study_link` and combine with the variant ID value in `variant.id`.

Also, can we please ensure that the table is sorted as it currently is in production - from highest to lowest p-value score?

**Comment by andrewhercules** [comment]
I have reviewed the current implementation on QA and it looks great! 👍 

![Screenshot 2020-02-24 at 11 30 55](https://user-images.githubusercontent.com/7490258/75149109-2375f580-56f9-11ea-8e3a-3de318e423aa.png)

@LucaFumis, can we please make a few updates:

1. By default, sort the data table by the Gene Prioritisation Score
2. In the Reported Trait column, link to the value to the study page on the Genetics Portal using the `evidence.variant2disease.study_link` value 
3. Create a tooltip with placeholder for the Gene Prioritisation Score (text will be confirmed once documentation is ready)
4. In the Variant column, display the RSID
5. Drop the Variant type column (Note: we were previously able to map the SO terms, but I wonder if the new data has broken this logic. @d0choa and I have discussed and if the logic has been broken, we can simply remove this column from the table)
6. For the OT Genetics Portal link in the Evidence Source column, hyperlink to the study-locus page in the Genetics Portal (e.g. https://genetics.opentargets.org/study-locus/GCST004610/19_1026478_G_T). To construct this link, please extract the study ID from `evidence.variant2disease.study_link` and combine with the variant ID value in `variant.id`.

**Comment by d0choa** [comment]
2 extra notes in case we resolve all the above issues:
- If we think we can rescue the full functionality of the variant type (5) it might be a good addition. If it's too complex, drop it as @andrewhercules suggested. That information is available when you land in the genetics portal.
- We can consider showing the publication field differently. Since it will be 1 or no publication in most of the cases, we can show <First author>, <year> (as shown in genetics portal), with the link to the paper

**Comment by d0choa** [comment]
Keep in mind there were issues before with the variant type resolver #757

**Comment by LucaFumis** [comment]
Yes exactly, the variant type seems a data issue - the field `evidence.evidence_codes_info` is `null`.

**Comment by LucaFumis** [comment]
Regarding point 4, it seems that for genetics portal data the rsid is not always available. I've updated the table to show when avail.

**Comment by d0choa** [comment]
The mantissa and exponent work is still pending It will finish the work started in #434 

A reference example:
https://qa.targetvalidation.org/evidence/ENSG00000146374/EFO_0009270?view=sec:genetic_association
